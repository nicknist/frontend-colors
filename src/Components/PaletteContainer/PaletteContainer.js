import React from 'react';
import Palette from '../Palette/Palette';

const PaletteContainer = ({ palettes, deletePalette, setPalettes }) => {
  return (
    <>
      {palettes.map(palette => <Palette
        key={palette.colors[0].color}
        name={palette.name}
        colors={palette.colors} deletePalette={deletePalette}
        setPalettes={setPalettes}
        palettes={palettes}
      />)}
    </>
  )
}

export default PaletteContainer;
