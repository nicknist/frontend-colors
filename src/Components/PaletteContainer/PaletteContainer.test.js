import React from 'react';
import { shallow } from 'enzyme';
import PaletteContainer from './PaletteContainer';

describe('Palette Container', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PaletteContainer
      palettes={[ {
          colors: [ { color: 'john' }],
          name: 'ben',
        }
      ]}
      deletePalette={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})
