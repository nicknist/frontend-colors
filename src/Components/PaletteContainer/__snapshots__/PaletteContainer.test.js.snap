// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette Container should match the snapshot 1`] = `
<Fragment>
  <Palette
    colors={
      Array [
        Object {
          "color": "john",
        },
      ]
    }
    deletePalette={[MockFunction]}
    key="john"
    name="ben"
    palettes={
      Array [
        Object {
          "colors": Array [
            Object {
              "color": "john",
            },
          ],
          "name": "ben",
        },
      ]
    }
  />
</Fragment>
`;
